python/lsst/meas/base/forcedPhotImage.py

```diff
@@ -143,6 +140,23 @@ def runDataRef(self, dataRef, psfCache=None):
         if psfCache is not None:
             exposure.getPsf().setCacheSize(psfCache)
         refCat = self.fetchReferences(dataRef, exposure)
+
+        measCat = self.run(exposure, refCat, refWcs, dataRef)
+
+        self.writeOutput(dataRef, measCat)
+
+    def run(self, exposure, refCat, refWcs, dataRef):
```

Comment:
As @yalsayyad pointed out on the ticket, please don't pass the dataRef to `run`.

If self.attachFootprints really needs a dataRef (I haven't looked through the stack to see how that is used) then I don't see an obvious alternative and it may be best to abandon this refactoring.

If `self.attachFootprints` doesn't actually need the dataRef (but can use something less butler-ish as a replacement) then the rest of the info can be passed into `run` as arguments: an id factory, an exposure ID and the dataId (the latter as an optional argument, since it's just used for logging).